### PR TITLE
Sync

### PR DESCRIPTION
**Pull request by aaronklaassen**
⚠️  Work in progress. A continuation of #1.

@mrmwiebe Sync-related commits can go on this `sync` branch. They'll show up in this PR and we can comment on shit.

https://trello.com/c/Qi5QO7so/27-sync

### CODE REVIEW

File: Assets/Scripts/System/Managers/OptionsManager.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        
+        //Load that JSON
+        string optionsFile = Application.dataPath + "/Options/winnitron_options.json";
+        Debug.Log("Loading options from " + optionsFile);
+        var O = GM.data.LoadJson(optionsFile);
+
+        //Widescreen
+        if (O ["launcher"] ["widescreen"] == "false") {
+            widescreen = false;
+        } else {
```
Comment (aaronklaassen) [inline Assets/Scripts/System/Managers/OptionsManager.cs]: Why does this work this way? Why do we skip loading options if `widescreen == true`?

File: Assets/Scripts/System/Managers/OptionsManager.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            // Load that JSON
+            // O = GM.data.LoadJson(contentPath + "/Options/winnitron_options.json");
```
Comment (aaronklaassen) [inline Assets/Scripts/System/Managers/OptionsManager.cs]: This isn't needed either, right? We should always load the same options file? (i.e. the one above, via `Application.dataPath`, not `contentPath`?)

File: Assets/Scripts/System/GM.cs [hunk 3]
```diff
@@ -20,6 +21,7 @@ public class GM : Singleton<GM> {
         options = GetComponent<OptionsManager> ();
 		state = GetComponent<StateManager> ();
         dbug = GetComponent<Dbug>();
+        sync = GetComponent<GameSync>();
```
Comment (aaronklaassen) [inline Assets/Scripts/System/GM.cs]: It's not, though. The GameSync component doesn't (or hasn't yet?) been initialized. So when trying to run the sync I get this:
screen shot 2016-10-17 at 9 01 27 pm

File: Assets/Scripts/System/Managers/OptionsManager.cs (hunk 1 above)
Comment (mrmwiebe) [inline Assets/Scripts/System/Managers/OptionsManager.cs]: Huh, I dunno.  I'll check it out and see why it's like that.

Comment (mrmwiebe) [inline Assets/Scripts/System/Managers/OptionsManager.cs]: There was some Git weirdness, not sure what happened, but if you look at the Develop branch, this bit of code isn't there.  Not sure where it came from.

File: Assets/Scripts/System/Managers/OptionsManager.cs (hunk 2 above)
Comment (mrmwiebe) [inline Assets/Scripts/System/Managers/OptionsManager.cs]: Again, take a look at the latest Develop branch OptionsManager.cs, pretty sure this bit of code isn't there.  Something got mixed in the last bit of Gittery.  Might need to do some of the ol' manual diff check.

File: Assets/Scripts/System/GM.cs (hunk 3 above)
Comment (aaronklaassen) [inline Assets/Scripts/System/GM.cs]: Problem solved. You are a hero, Marlon. 😍

Comment (mrmwiebe) [inline Assets/Scripts/System/GM.cs]: 😉

**Comment by aaronklaassen** [comment]
Manually merged.

**Pull request closed without merging**